internal/dag/ingress_processor.go

```diff
@@ -209,6 +215,24 @@ func route(ingress *networking_v1.Ingress, path string, service *Service, client
 	}
 
 	r.PathMatchCondition = &PrefixMatchCondition{Prefix: path}
+
+	// If we have a wildcard match, add a header match regex rule to match the
+	// hostname so we can be sure to only match one DNS label. This is required
+	// as Envoy's virtualhost hostname wildcard matching can match multiple
+	// labels. This match ignores a port in the hostname in case it is present.
+	if strings.HasPrefix(host, "*.") {
```

Comment:
Should we validate the only a single `*` was added? Meaning if someone entered: `*.foo.*.com` it should get thrown out?

Reply:
`Ingress` objects seem to already have validation set up so I skipped doing that in Contour:

The following gets an error when applied:

```
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: test
spec:
  rules:
    - host: "*.projectcontour.*.io"
      http:
        paths:
        - pathType: Prefix
          path: "/"
          backend:
            service:
              name: echo
              port:
                number: 80
```

Error:

```
% kubectl apply -f /tmp/ingress.yaml
The Ingress "test" is invalid: spec.rules[0].host: Invalid value: "*.projectcontour.*.io": a wildcard DNS-1123 subdomain must start with '*.', followed by a valid DNS subdomain, which must consist of lower case alphanumeric characters, '-' or '.' and end with an alphanumeric character (e.g. '*.example.com', regex used for validation is '\*\.[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```

Reply:
👍